Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/VaultManager.swift

```diff
@@ -40,24 +42,80 @@ extension PrimerHeadlessUniversalCheckout {
             try self.validate()
         }
         
-        private func validate() throws {
-            guard PrimerAPIConfigurationModule.decodedJWTToken != nil,
-                  PrimerAPIConfigurationModule.apiConfiguration != nil
-            else {
-                let err = PrimerError.uninitializedSDKSession(userInfo: nil, diagnosticsId: UUID().uuidString)
-                ErrorHandler.handle(error: err)
-                throw err
-            }
+        internal func validateAdditionalDataSynchronously(vaultedPaymentMethodId: String, vaultedPaymentMethodAdditionalData: PrimerVaultedPaymentMethodAdditionalData) -> [Error]? {
+            self.vaultedPaymentMethodAdditionalData = nil
+            var errors: [Error] = []
             
-            guard let customerId = PrimerAPIConfigurationModule.apiConfiguration?.clientSession?.customer?.id else {
-                let err = PrimerError.invalidClientSessionValue(
-                    name: "customer.id",
-                    value: nil,
-                    allowedValue: "string",
-                    userInfo: nil,
+            guard let vaultedPaymentMethod = self.vaultedPaymentMethods?.first(where: { $0.id == vaultedPaymentMethodId }) else {
+                let err = PrimerError.invalidVaultedPaymentMethodId(
+                    vaultedPaymentMethodId: vaultedPaymentMethodId,
+                    userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
                     diagnosticsId: UUID().uuidString)
                 ErrorHandler.handle(error: err)
-                throw err
+                errors.append(err)
+                return errors
+            }
+            
+            if vaultedPaymentMethod.paymentMethodType == "PAYMENT_CARD" {
+                let cardNetwork = CardNetwork(cardNetworkStr: vaultedPaymentMethod.paymentInstrumentData.network ?? "")
```

Comment:
@EvansPie I believe we should use `paymentInstrumentData.binData.network` here.

Reply:
Done